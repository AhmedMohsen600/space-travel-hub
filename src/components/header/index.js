import { NavLink, Outlet } from 'react-router-dom';
import {
  Container,
  Nav,
  Group,
  TextLink,
  Logo,
  Image,
  Title,
} from './styles/header';
import { links } from '../../fixtures/links';
import './styles/nav.css';

export default function Header() {
  return (
    <>
      <Container>
        <Nav>
          <Logo>
            <Image src="/travel-hub.png" />
            <Title>Space Travel&apos;s hub</Title>
          </Logo>
          <Group>
            {links.map(({ url, text }) => (
              <TextLink key={url}>
                <NavLink activeclassname="active" className="link" to={url}>
                  {text}
                </NavLink>
              </TextLink>
            ))}
          </Group>
        </Nav>
      </Container>
      <Outlet />
    </>
  );
}
