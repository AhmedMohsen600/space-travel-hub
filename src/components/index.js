export { default as Header } from './header';
export { default as Table } from './table';
export { default as Rocket } from './rocket';
