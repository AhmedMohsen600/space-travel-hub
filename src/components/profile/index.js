import {
  BookedMissionsWrapper,
  MissionsWrapper,
  RocketsWrapper,
  Mission,
  Rocket,
} from './styles/profile';

const Profile = ({ reservedRockets, reservedMissions }) => {
  return (
    <>
      <BookedMissionsWrapper>
        <MissionsWrapper>
          <h2>My Missions</h2>
          {reservedMissions.map((mission) => (
            <Mission key={mission.mission_id}>{mission.mission_name}</Mission>
          ))}
        </MissionsWrapper>
        <RocketsWrapper>
          <h2>My Rockets</h2>
          {reservedRockets.map((rocket) => (
            <Rocket key={rocket.rocket_id}>{rocket.rocket_name}</Rocket>
          ))}
        </RocketsWrapper>
      </BookedMissionsWrapper>
    </>
  );
};

export default Profile;
