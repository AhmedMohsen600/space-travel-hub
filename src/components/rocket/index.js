import { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { fetchRockets, reservation } from '../../features/rockets/rocketsSlice';
import {
  Container,
  RocketName,
  RocketHolder,
  RocketDescription,
  RocketInfoHolder,
  RocketImage,
  ImageHolder,
  ReserveBtn,
  Inner,
  Loading,
  Group,
  Mark,
} from './styles/rocket';

export default function Rocket() {
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(fetchRockets());
  }, [dispatch]);
  const rockets = useSelector((state) => state.rockets.data);
  const loading = useSelector((state) => state.rockets.loading);

  return (
    <Container>
      <Inner>
        {loading ? (
          <Loading>Loading</Loading>
        ) : (
          rockets.map((rocket) => (
            <RocketHolder key={rocket.id}>
              <ImageHolder>
                <RocketImage src={rocket.flickr_images} />
              </ImageHolder>
              <RocketInfoHolder>
                <RocketName>{rocket.rocket_name}</RocketName>
                <Group>
                  {rocket.reserved && <Mark>reserved</Mark>}
                  <RocketDescription>{rocket.description}</RocketDescription>
                </Group>
                <ReserveBtn
                  reserved={rocket.reserved}
                  onClick={() => dispatch(reservation(rocket.id))}
                >
                  {rocket.reserved ? 'Cancel Reservation' : 'Reserve Rocket'}
                </ReserveBtn>
              </RocketInfoHolder>
            </RocketHolder>
          ))
        )}
      </Inner>
    </Container>
  );
}
