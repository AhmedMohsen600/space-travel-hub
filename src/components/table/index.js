import { useSelector, useDispatch } from 'react-redux';
import { useEffect } from 'react';
import {
  fetchMissions,
  changeMission,
} from '../../features/missions/missionsSlice';
import {
  MissionsWrapper,
  MainHeadings,
  TableBody,
  MissionStatus,
  JoinMission,
  MissionStatusWrapper,
} from './styles/table';

const Table = () => {
  const dispatch = useDispatch();
  const missions = useSelector((state) => state.missions.missionsArr);
  useEffect(() => {
    if (missions.length) return;
    dispatch(fetchMissions());
  }, [dispatch, missions.length]);
  return (
    <MissionsWrapper>
      <MainHeadings>
        <tr>
          <th>Mission</th>
          <th>Description</th>
          <th>Status</th>
          <th aria-label="Join or leave mission" />
        </tr>
      </MainHeadings>
      <TableBody>
        {missions.map((mission) => (
          <tr key={mission.mission_id}>
            <MissionStatusWrapper>{mission.mission_name}</MissionStatusWrapper>
            <td>{mission.description}</td>
            <MissionStatusWrapper>
              <MissionStatus active={mission.activeMission} type="button">
                {mission.activeMission ? 'Active Member' : 'NOT A MEMBER'}
              </MissionStatus>
            </MissionStatusWrapper>
            <MissionStatusWrapper>
              <JoinMission
                active={mission.activeMission}
                onClick={() => dispatch(changeMission(mission.mission_id))}
                type="button"
              >
                {mission.activeMission ? 'Leave Mission' : 'Join Mission'}
              </JoinMission>
            </MissionStatusWrapper>
          </tr>
        ))}
      </TableBody>
    </MissionsWrapper>
  );
};

export default Table;
