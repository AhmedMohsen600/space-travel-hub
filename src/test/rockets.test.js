import { render } from '@testing-library/react';
import { useSelector } from 'react-redux';
import { Rocket } from '../components';

jest.mock('react-redux');

describe('Rockets test', () => {
  test('Should test rockets Componenet', () => {
    useSelector.mockReturnValue([
      {
        id: '123',
        flickr_images:
          'https://imgur.com/DaCfMsj.jpg,https://imgur.com/azYafd8.jpg',
        rocket_name: 'Falcon 1',
        description: 'Test',
        reserved: !true,
      },
    ]);
    const tree = render(<Rocket />);
    expect(tree).toMatchSnapshot();
  });
});
