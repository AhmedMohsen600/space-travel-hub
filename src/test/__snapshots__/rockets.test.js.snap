// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rockets test Should test rockets Componenet 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="rocket__Container-sc-18ebhcu-0 ddltMC"
      >
        <ul
          class="rocket__Inner-sc-18ebhcu-1 hPMSEV"
        >
          <div
            class="rocket__Loading-sc-18ebhcu-10 fCybjE"
          >
            Loading
          </div>
        </ul>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="rocket__Container-sc-18ebhcu-0 ddltMC"
    >
      <ul
        class="rocket__Inner-sc-18ebhcu-1 hPMSEV"
      >
        <div
          class="rocket__Loading-sc-18ebhcu-10 fCybjE"
        >
          Loading
        </div>
      </ul>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
